### PR TITLE
336: Update github workflows

### LINKED ISSUE #336: Pipeline deploy to Prod and Test
**Describe the Task**
Use example in Lucy-web

### PR DESCRIPTION
**Pull request by NickPhura**
# Overview

This PR includes the following proposed change(s):

### Commit 1
- Update github workflows to work across dev, test and prod, and to cleanup PR artifacts better.
- Added a new database dockerfile, to distinguish between PR builds (which run migrations + seeding) and dev/test/prod builds (which only run migrations)

### Commit 2
- Had to add some extra type casts to satisfy the openshfit api build.  Was unable to re-produce locally which is odd.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Via many PR builds.

## Screenshots

Please add any relevant UI screenshots if applicable.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: .github/workflows/clean-pr.yml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+          echo GIt Event Name: ${{ github.event_name }}
+          echo GIt Event Action: ${{ github.event.action }}
+          echo Git Title: ${{ github.event.pull_request.title }}
+          echo Git Labels: "$LABELS"
+
+  clean:
+    name: Clean Build and Deployment Artifacts in the Dev and Tools OpenShift environments
+    runs-on: ubuntu-16.04
+    # Only clean if the PR that closed was marked as a PR-DEPLOY
+    if: ${{ contains(github.event.pull_request.labels.*.name, 'PR-DEPLOY') || contains(github.event.pull_request.title, 'PR-DEPLOY') }}
```
Comment (sdevalapurkar) [inline .github/workflows/clean-pr.yml:41]: nice! much cleaner

File: .github/workflows/deploy-pr.yml [hunk 2]
```diff
@@ -116,15 +126,15 @@ jobs:
       - name: Build API Image
         working-directory: './api/.pipeline/'
         run: |
-          # npm install
+          #npm install
```
Comment (sdevalapurkar) [inline .github/workflows/deploy-pr.yml]: can we remove this?

Comment (NickPhura) [reply .github/workflows/deploy-pr.yml]: yup!

**Review by sdevalapurkar — APPROVED** [review]
LGTM :)

**Pull request merged**